include/arch/arm64/cpu.h

```diff
@@ -206,5 +206,9 @@
 #define L1_CACHE_SHIFT		(6)
 #define L1_CACHE_BYTES		BIT(L1_CACHE_SHIFT)
 #define ARM64_CPU_INIT_SIZE	L1_CACHE_BYTES
+/* offset (bytes) of SP field in struct arm64_cpu_init */
+#define ARM64_INIT_TB_SP_OFFSET		0
+/* offset (bytes) of mpid field in struct arm64_cpu_init */
+#define ARM64_INIT_TB_MPID_OFFSET	24
```

Comment:
Argh, no. This is 100% going to break if the struct is reworked.

Reply:
We do have an `offsetof` macro

Reply:
Thanks for your comments, but I find we can't use offsetof macro in assembly language, so I would like to move MPID filed to be fixed in secondary variable in struct of arm64_cpu_init (the first is SP), and add annotation in struct definition to address this issue.